### ISSUE TITLE
ZMQ topic prefix matching issue

### ISSUE THREAD

**Issue by donsander** [iotaledger/iri#741]
Subscribing to the "tx" topic with ZMQ will also subscribe to the new "tx_trytes" topic, and each new transaction will be received twice.  This is unwanted behavior, and has broken some live tangle visualizers. 

I believe this is due to both topics having a "tx" prefix.  By design ZMQ will perform prefix matching on topics so you can subscribe to multiple topics at once.  This makes it difficult to only subscribe to the "tx" topic alone because it will also match "tx_trytes".

One hacky workaround is to subscribe to "tx " instead of "tx".  Another is to use message filtering option available for ZMQ clients but this is not ideal.  

Perhaps the topic names should be changed to "tx_bytes" and "tx_trytes".  Or to retain some backwards compatibility (which was already lost with the last update) just change the name of the "tx_trytes" topic to something like "tryts".

**Comment by alon-e** [comment]
@donsander thanks for bringing this up.
>has broken some live tangle visualizers.

would be great if you also give more context to what implementation it broke and who are the developers, so we can more extensively test in future - for example, in #530 we asked known ZMQ user to test it: https://github.com/iotaledger/iri/pull/540#issuecomment-378636008

>One hacky workaround is to subscribe to "tx " instead of "tx". Another is to use message filtering option available for ZMQ clients but this is not ideal.

[I agree :)](https://github.com/alon-e/iota-ctps/commit/39bcb8ee784a81deee759ca1a1c6a0339d51d58e
), however, until the issue is resolved this workaround can be used.

I want to remind users that ZMQ [is still WIP,](https://github.com/iotaledger/iri/tree/dev/src/main/java/com/iota/iri/zmq) and will break functionality sometimes, topic names added and changed etc - that's why it is disabled by default. however, we'll try to minimize this and communicate changes more clearly.

**Comment by donsander** [comment]
@alon-e The change caused issues with the 3D tangle viewer I created at live.iotaknot.com - the backend is node.js connecting to an IRI instance. The additional messages from "tx_trytes" were not expected, and for obvious reasons parsing the message would cause issues. If you have any more questions I can answer them.  I have since used the "tx " workaround so everything is working as it should now.

As for ZMQ being WIP, I am aware, and this issue report is just a heads up.  I am thankful that #530 has resolved an issue that I have been dealing with for a long time which required me to restart my IRI instance constantly during high traffic times.  My visualizer is now stable again!
